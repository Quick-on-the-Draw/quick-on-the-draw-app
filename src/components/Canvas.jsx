import React, { useState, useRef } from 'react';
import CanvasDraw from 'react-canvas-draw';
import BrushSize from './Buttons/BrushSize';
import ColorPallette from './Buttons/ColorPallette';
import style from './Canvas/Canvas.module.css';

const Canvas = () => {
  const [brushColor, setBrushColor] = useState('#000000');
  const [brushSize, setBrushSize] = useState(1);

  const canvasRef = useRef();
  const canvasProps = {
    lazyRadius: 0,
    brushColor: brushColor,
    brushRadius: brushSize,
    hideInterface: true,
    hideGrid: true,
  };

  const handleColorChange = (color) => {
    setBrushColor(color);
  };

  return (
    <div>
      <div className={style.canvasContainer}>
        <CanvasDraw
          {...canvasProps}
          className={style.mainCanvas}
          ref={canvasRef}
        />
      </div>

      <div>
        <ColorPallette
          pickColor={handleColorChange}
          setBrushColor={setBrushColor}
        />
        <BrushSize setBrushSize={setBrushSize} />
      </div>
      <button onClick={() => canvasRef.current.undo()}>Undo</button>
      <button onClick={() => canvasRef.current.clear()}>Clear Canvas</button>
    </div>
  );
};

export default Canvas;
