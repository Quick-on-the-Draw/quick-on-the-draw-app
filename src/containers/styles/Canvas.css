@tailwind base;
@tailwind components;
@tailwind utilities;

.main {
    @apply bg-scroll
    bg-center
    bg-cover
    h-screen;
}

.container {
    @apply flex 
    flex-col 
    justify-center 
    items-center;
}

.colorContainer {
    @apply grid grid-cols-3 gap-2;
}
.brushDiv {
    @apply col-start-3;
}
.imgDiv {
    @apply col-span-2;
}

.colorDiv {
    @apply col-start-3;
}

.sideContainer {
    @apply flex
    flex-row-reverse;
}

.colorPalette {
    @apply flex
    flex-col;
}

.mainCanvas {
    @apply h-12
    w-12
    border-8
    border-yellow
    border-dashed
    rounded-lg
    shadow-2xl
    flex
    content-center
    justify-center
    m-2;
}

.canvasContainer {
}

.cactus {
    @apply h-full
  w-full;
}

.title {
    @apply h-60
  w-64;
}
