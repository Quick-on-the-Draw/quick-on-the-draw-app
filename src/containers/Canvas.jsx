import React, { useState, useRef } from 'react';
import CanvasDraw from 'react-canvas-draw';
import BrushSize from '../components/Buttons/BrushSize';
import ColorPalette from '../components/Buttons/ColorPalette';
import ControlButtons from '../components/Buttons/ControlButtons';
import Timer from '../components/Timer/Timer';
import './styles/Canvas.css';

const Canvas = () => {
    const [brushColor, setBrushColor] = useState('#000000');
    const [brushSize, setBrushSize] = useState(1);
    // const [loading, setLoading] = useState(true);

  const canvasRef = useRef();
  const canvasProps = {
    lazyRadius: 0,
    brushColor: brushColor,
    brushRadius: brushSize,
    hideInterface: true,
    hideGrid: true,
    canvasWidth: 700,
    canvasHeight: 700,
  };

  const handleColorChange = (color) => {
    setBrushColor(color);
  };

  const handleUndo = () => {
    canvasRef.current.undo();
  };

  const handleClear = () => {
    canvasRef.current.clear();
  };

  return (
    <div className="container">
      <Timer />
      <div>
        <CanvasDraw {...canvasProps} className="mainCanvas" ref={canvasRef} />
      </div>
      <div>
        <ColorPalette
          pickColor={handleColorChange}
          setBrushColor={setBrushColor}
        />
        <BrushSize setBrushSize={setBrushSize} />
        <ControlButtons handleUndo={handleUndo} handleClear={handleClear} />
      </div>
    </div>
  );
};

export default Canvas;
