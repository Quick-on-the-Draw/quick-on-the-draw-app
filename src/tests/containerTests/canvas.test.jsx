import { render, screen } from '@testing-library/react';
import Canvas from '../../containers/Canvas';
import React from 'react';

describe.skip('Canvas page tests', () => {
    it('looks for the navbar on the Canvas page', async () => {
        render(<Canvas />);

        const navbar = await screen.findByText('Home');
        expect(navbar).toMatchSnapshot();
    });
});
