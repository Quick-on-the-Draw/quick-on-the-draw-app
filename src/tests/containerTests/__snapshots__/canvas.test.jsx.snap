// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas page tests looks for the navbar on the Canvas page 1`] = `
<img
  alt="Save Button"
  class="save"
  src="save.png"
/>
`;
